### DISCUSSION TITLE
CodeForces Ödevi

### DISCUSSION THREAD

**Discussion by Erdem30** [sonsuzus/sonsuzus#512] [Soru Cevap]
Boris Notkin melodiler besteler. Bunları, her notanın 0'dan 127'ye kadar bir tam sayı olarak kodlandığı bir nota dizisi olarak gösterir. İki nota a ve b arasındaki aralık |a−b|

yarım tona eşittir.

Boris, her iki bitişik nota arasındaki aralık 5
yarım ton veya 7

yarım ton ise melodiyi mükemmel olarak kabul eder.

Son melodilerini besteledikten sonra, eserlerinin koleksiyonunu size coşkuyla gösterir. Boris Notkin'in melodilerinin mükemmel olup olmadığını anlamasına yardımcı olun.
Giriş

İlk satırda bir tam sayı t
(1≤t≤1000

) bulunur — melodi sayısı.

Her melodi iki satırla tanımlanır.

İlk satırda bir tam sayı n
(2≤n≤50

) bulunur — melodideki nota sayısı.

İkinci satır n
tam sayı a1,a2,…,an (0≤ai≤127

) içerir — melodinin notaları.
Çıktı

Her melodi için, mükemmel ise "EVET" çıktısı alın; aksi takdirde "HAYIR" çıktısı alın.

Cevabı herhangi bir durumda (üst veya alt) çıktı olarak alabilirsiniz. Örneğin, "yEs", "yes", "Yes" ve "YES" dizeleri olumlu yanıtlar olarak tanınacaktır. Örnek
Giriş
Kopyala

8
2
14 09
2
17 10
3
76 83 88
8
38 45 38 80 85 92 99 106
5
63 58 65 58 65
8
117 124 48 53 48 43 54 49
5
95 102 107 114 121
10
72 77 82 75 70 75 68 75 68 75

Çıktı
Kopyala

EVET
EVET
EVET
HAYIR
EVET
HAYIR
EVET
EVET

**Comment by Erdem30** [comment]
rows = int(input("Melodi sayısını giriniz (1-1000 arası): "))

array_2d = []


for i in range(rows):
    cols = int(input(f"{i + 1}. melodideki nota sayısını giriniz (2-50 arası): "))
    row = []
    
    for j in range(cols):
        element = int(input(f"{i + 1}. melodideki {j + 1}. notayı girin (1 - 127 arası): "))
        
        row.append(element)
    
    array_2d.append(row)


def check_difference_row(array_2d):
    for row in array_2d:

        all_valid = True
        

        for i in range(len(row) - 1):

            if abs(row[i] - row[i + 1]) != 5 and abs(row[i] - row[i + 1]) != 7:
                all_valid = False 
                break  
        
        if all_valid:
            print("EVET")
        else:
            print("HAYIR")

check_difference_row(array_2d)

**Reply by sonsuzus** [reply]
kodunu code içine alırsan okunurluğu artar. Ayrıca çeviri ve soru eklemesi de kötü olmuş. Onları da düzenlersin.

Bir de kod senin mi yapay zekanın mı konuşuruz. :)
